src/main/java/cubicchunks/worldgen/dependency/RegionDependency.java

```diff
@@ -36,40 +36,24 @@
 public class RegionDependency implements Dependency {
 	
 	private GeneratorStage targetStage;
-	
-	private Set<Long> requiredCubes;
-	
 	private Set<Requirement> requirements;
-
-	private int xLow;
-	private int xHigh;
-	private int yLow;
-	private int yHigh;
-	private int zLow;
-	private int zHigh;
 	
-	
-	public RegionDependency(GeneratorStage targetStage, int radius) {
+	public RegionDependency(Cube cube, GeneratorStage targetStage, int radius) {
```

Comment:
Currently its per cube. I considered it to be somewhat more clean, because this permits stateaware dependencies per cube. But if you prefer singletons, I can change it.